src/Components/Nav/Nav.js

```diff
@@ -1,17 +1,46 @@
 import React, { Component } from 'react';
-import { Link } from 'react-router-dom';
+import { Link, withRouter } from 'react-router-dom';
 import InnerNav from './InnerNav/InnerNav';
 import NAV_DATA from './navData';
 import './Nav.scss';
 
 class Nav extends Component {
+  constructor() {
+    super();
+    this.state = {
+      hasToken: false,
+    };
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.location.pathname !== prevProps.location.pathname) {
+      if (
+        window.sessionStorage.AccessToken ||
+        window.localStorage.AccessToken
+      ) {
+        this.setState({
+          hasToken: true,
+        });
+      }
+    }
+  }
+
+  setLogout = () => {
+    window.sessionStorage.removeItem('accessToken');
+
+    this.setState({
+      hasToken: false,
+    });
+  };
+
   render() {
+    const { hasToken } = this.state;
```

Comment:
```suggestion
    const hasToken =  window.sessionStorage.AccessToken ||
        window.localStorage.AccessToken
```
굳이 state로 괸리할 필요는 없어보이네요! 이해 안되시면 라운지에서 질문해주세요! 그리고 session이나 local 둘 중 어디에 저장할지 정하고 한 storage 에서만 확인하는게 좋아보입니다!

Reply:
반영하겠습니다~!